plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchDefaultExtensionModule.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.elasticsearch;
+
+import com.google.inject.Binder;
+import io.airlift.configuration.AbstractConfigurationAwareModule;
+import io.trino.plugin.elasticsearch.aws.ElasticsearchAwsModule;
+
+import static io.airlift.configuration.ConditionalModule.conditionalModule;
+import static io.trino.plugin.elasticsearch.ElasticsearchConfig.Security.AWS;
+import static java.util.function.Predicate.isEqual;
+
+public class ElasticsearchDefaultExtensionModule
+        extends AbstractConfigurationAwareModule
+{
+    @Override
+    protected void setup(Binder binder)
+    {
+        install(conditionalModule(
+                ElasticsearchConfig.class,
+                config -> config.getSecurity()
+                        .filter(isEqual(AWS))
+                        .isPresent(),
+                new ElasticsearchAwsModule()));
```

Comment:
Why is this a "default extension". The AWS authentication is a built-in feature of the elasticsearch connector.

Reply:
I initially let it as part of the `ElasticsearchConnectorModule` and changed to this to apply [this comment](https://github.com/trinodb/trino/pull/8603#discussion_r672907640). Should I move it back to `ElasticsearchConnectorModule`?

Reply:
That way we are able to replace the implementation of AWS authentication.

Reply:
@kokosing since we're using an Optional binder and providing a default implementation to it in `ElasticsearchAwsModule` it'd still be possible to install that module in `ElasticsearchConnectorModule` and allow others to override the optional by calling `setBinding()`. 

I kept this separate "default extension" module for now but please let me know if I should remove it or not.

cc @martint